maps/src/main/java/org/odk/collect/maps/MarkerIconDescription.kt

```diff
@@ -0,0 +1,53 @@
+package org.odk.collect.maps
+
+import android.graphics.Color
+import org.odk.collect.shared.strings.StringUtils
+import java.util.Locale
+
+class MarkerIconDescription @JvmOverloads constructor(
+    val iconDrawableId: Int,
+    private val color: String? = null,
+    private val symbol: String? = null
+) {
+    fun getColor(): Int? = try {
+        color?.let {
+            var sanitizedColor = if (color.startsWith("#")) {
+                color
+            } else {
+                "#$color"
+            }
+
+            if (sanitizedColor.length == 4) {
+                sanitizedColor = shorthandToLonghandHexColor(sanitizedColor)
+            }
+
+            Color.parseColor(sanitizedColor)
+        }
+    } catch (e: Throwable) {
+        null
+    }
+
+    fun getSymbol(): String? = symbol?.let {
+        if (it.isBlank()) {
+            null
+        } else {
+            StringUtils.firstCharacterOrEmoji(it).uppercase(Locale.US)
+        }
+    }
+
+    override fun hashCode(): Int {
```

Comment:
Not seeing why `hashCode` is overridden here? I get that there would need to be a custom implementation to maintain the behaviour we lose from moving away from `data class`, but I can't find anything that requires that behaviour. Additionally, even those this doesn't breach the `hashCode`/`equals` relationship (objects that are equal would have the same `hashCode` here), it does feel weird to have a smart `hashCode` implementation but leave `equals` using basic object equality as it massively increases the cases where two objects can have the same `hashCode` but be "unequal".

Reply:
I've removed this and all tests seem to be passing.